src/onnx/parse_multinomial.cpp

```diff
@@ -0,0 +1,64 @@
+#include <migraphx/onnx/op_parser.hpp>
+#include <migraphx/onnx/checks.hpp>
+#include <migraphx/ranges.hpp>
+#include <migraphx/instruction.hpp>
+#include <migraphx/make_op.hpp>
+#include <random>
+
+namespace migraphx {
+inline namespace MIGRAPHX_INLINE_NS {
+namespace onnx {
+
+struct parse_multinomial : op_parser<parse_multinomial>
+{
+    std::vector<op_desc> operators() const { return {{"Multinomial"}}; }
+
+    instruction_ref parse(const op_desc& /*opd*/,
+                          const onnx_parser& parser,
+                          const onnx_parser::node_info& info,
+                          std::vector<instruction_ref> args) const
+    {
+        int dtype = 6; // 6=in32, 7=int64
+        if(contains(info.attributes, "dtype"))
+            dtype = parser.parse_value(info.attributes.at("dtype")).at<int>();
+
+        size_t sample_size = 1;
+        if(contains(info.attributes, "sample_size"))
+            sample_size = parser.parse_value(info.attributes.at("sample_size")).at<int>();
+
+        float seed = static_cast<float>(
+            std::chrono::high_resolution_clock::now().time_since_epoch().count());
+        if(contains(info.attributes, "seed"))
+            seed = parser.parse_value(info.attributes.at("seed")).at<float>();
+
+        // Compute cumulative density function
+        auto maxes =
```

Comment:
Where do these algos come from?

Reply:
These accomplish the same computation found in MultinomialCompute up until the the Generate each sample comment from [here](https://github.com/microsoft/onnxruntime/blob/8d737f977056444a307f1b7f0bcd402fba62d790/onnxruntime/core/providers/cpu/generator/random.cc).

Reply:
Just one last minor suggestion: can you add a comment on the source for this algorithm? For exmaple in onnx runtime source code, they mention tensorflow, and we can mention we followed a similar variation.